x-pack/legacy/plugins/apm/public/components/app/Settings/AgentConfigurations/AddEditFlyout/ServiceSection.tsx

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { EuiTitle, EuiSpacer, EuiFormRow } from '@elastic/eui';
+import React from 'react';
+import { i18n } from '@kbn/i18n';
+import { SelectWithPlaceholder } from '../../../../shared/SelectWithPlaceholder';
+import { useFetcher } from '../../../../../hooks/useFetcher';
+import { ENVIRONMENT_NOT_DEFINED } from '../../../../../../common/environment_filter_values';
+import { Config } from '../index';
+import { callApmApi } from '../../../../../services/rest/callApmApi';
+const t = (id: string, defaultMessage: string) =>
+  i18n.translate(`xpack.apm.settings.agentConf.flyOut.serviceSection.${id}`, {
+    defaultMessage
+  });
+
+const SELECT_PLACEHOLDER_LABEL = `- ${t('selectPlaceholder', 'Select')} -`;
+
+interface Props {
+  selectedConfig: Config | null;
+  environment?: string;
+  setEnvironment: (env: string) => void;
+  serviceName?: string;
+  setServiceName: (env: string) => void;
+}
+
+export function ServiceSection({
+  selectedConfig,
+  environment,
+  setEnvironment,
+  serviceName,
+  setServiceName
+}: Props) {
+  const { data: serviceNames = [], status: serviceNamesStatus } = useFetcher(
+    () => {
+      return callApmApi({
```

Comment:
Hmm... I thought about it but I think it's okay since it's only in-memory caching, so whenever the user comes back to Kibana they will see fresh data. It is also not data that changes very often. Most users have long running application names. Do you think it's likely that someone changes the service name or adds a new service while being on this page?

Reply:
Hmm, maybe. I can imagine they want to configure a service around the time it starts sending data. Then again, they will probably just refresh the page anyway.